design/ErrorHandling.md

```diff
@@ -91,76 +133,39 @@ Using any invalid object in a WebGPU operation produces a validation error.
-
-If an application uses recoverable allocation, the implementation will still generate error log entries:
-a `"recoverable-out-of-memory"` error for the object creation, and `"validation-error"`s for any subsequent uses of the invalid object.
-The application may need to understand whether such error log entries were part of a recovered allocation (e.g. to avoid sending telemetry for those errors).
-To facilitate this filtering, the handle to the invalid object (including "expando" JavaScript properties) is attached to the error log entry (see above).
-
-### Recoverable errors in object creation
+If an application wants to allocate with fatal out-of-memory, it uses createBuffer/createTexture.
+Just like with any the creation of smaller objects, an out-of-memory condition will be treated as a fatal error: the device is lost.
 
-A recoverable error is exposed as a `WebGPUObjectStatusQuery`.
-
-```
-// (Exact form/type subject to change.)
-typedef Promise<WebGPUObjectStatus> WebGPUObjectStatusQuery;
-
-typedef (WebGPUBuffer or WebGPUTexture) StatusableObject;
-
-partial interface WebGPUDevice {
-    WebGPUObjectStatusQuery getObjectStatus(StatusableObject object);
-};
-```
-
-A concrete example: When creating a buffer, the following logic applies:
-
- - `createBuffer` returns a `WebGPUBuffer` object `buffer` immediately.
- - A `WebGPUObjectStatusQuery` can be obtained by calling `device.getObjectStatus(buffer)`.
-   At a later time, that query resolves to a `WebGPUObjectStatus` that is one of:
-    - Creation succeeded (`"valid"`).
-    - Creation encountered a recoverable error (`"out-of-memory"`).
-      (The application can then choose to retry a smaller allocation of a *new* `WebGPUBuffer`.)
-    - Creation encountered another type of error out of the control of the application (`"invalid"`).
-
-Regardless of any recovery efforts the application makes, if creation fails,
-the resulting object is invalid (and subject to error propagation).
-
-Checking the status of a `WebGPUBuffer` or `WebGPUTexture` is **not** required.
-It is only necessary if an application wishes to recover from recoverable errors such as out of memory.
-(If it does, it is responsible for avoiding using the invalid object.)
+The `tryCreate*` entry points return a Promise.
+ - If creation succeeds, the Promise resolves to a *valid* `GPUBuffer`/`GPUTexture`.
+ - If there is a validation error, the Promise resolves to an *invalid* `GPUBuffer`/`GPUTexture` (and produces a log entry).
+ - If the resource allocation runs out of memory, the Promise rejects.
 
 ## Open Questions and Considerations
 
- - WebGPU could guarantee that objects such as `WebGPUQueue` and `WebGPUFence` can never be errors.
-   If this is true, then the only synchronous API that needs special casing is buffer mapping, where `mapping` is always `null` for an invalid `WebGPUBuffer`.
+ - WebGPU could guarantee that objects such as `GPUQueue` and `GPUFence` can never be errors.
```

Comment:
The device can be created in an already "lost" state in which case the queue and the fence are errors,